src/single-post.ejs

```diff
@@ -0,0 +1,622 @@
+          }
+          if (imageInfo) {
+            imageInfo.map((el, index) => {
+              dotsHtml.innerHTML += `
+                         <span class="dot" onclick="currentSlide(${
+                           index + 1
+                         })"></span>
+              `;
+            });
+            imageInfo.map((el, index) => {
+              sliderHtml.innerHTML += `
+              <div class="mySlides fade">
+                    <div class="numbertext">${index + 1} / ${
+                imageInfo.length
+              }</div>
+                    <img
+                      src="${el.url}"
+                      alt="${el.alt}""
+                    />
+                  </div>
+
+              `;
+            });
+
+            imageInfo.map((el, index) => {
+              thumbNailHtml.innerHTML += `<img onclick="currentSlide(${
+                index + 1
+              })" src="${el.url}" alt="${el.alt}" class="thumb-nail__img" />`;
+            });
+          }
+
+          if (staticInfo) {
+            staticInfo.map(
+              (el) =>
+                (staticDataHtml.innerHTML += `
+              <div class="static-data">
+                    <img src="${el.icon.image_url_light}" class="static-data__img" />
+                    <span class="static-data__title"> ${el.title}</span>
+                  </div>
+              `)
+            );
+          }
+          if (tagsInfo) {
+            tagsInfo.map((el) => {
+              tagHtml.innerHTML += `<button class="tag">${el.text}</button>`;
+            });
+          }
+
+          if (navbarInfo) {
+            navbarInfo.parent_items.map((el) => {
+              navbarHtml.innerHTML += `
+              <li class="post-page__navbar-item">
+              <a>${el.title}</a>
+              <img class="post-page__navbar-icon" src="<%=require(
+              "./icons/svgs/arrow-left.svg" ).default%>"/>
+            </li>
+              `;
+            });
+
+            navbarHtml.innerHTML += `<span class="current_page_title">${navbarInfo.current_page_title}</span>`;
+          }
+        })
+        .catch((err) => console.log(err));
+
+      var slideIndex = 1;
+      showSlides(slideIndex);
+      function plusSlides(n) {
+        showSlides((slideIndex += n));
+      }
+      function currentSlide(n = 1) {
+        showSlides((slideIndex = n));
+      }
+      function showSlides(n = 1) {
+        var i;
+        var slides = document.getElementsByClassName("mySlides");
+        var dots = document.getElementsByClassName("dot");
+        if (slides && slides.length > 0) {
+          if (n > slides.length) {
+            slideIndex = 1;
+          }
+          if (n < 1) {
+            slideIndex = slides.length;
+          }
+          for (i = 0; i < slides.length; i++) {
+            slides[i].style.display = "none";
+          }
+          for (i = 0; i < dots.length; i++) {
+            dots[i].className = dots[i].className.replace(" active", "");
```

Comment:
Could've used classList and toggle

Reply:
Also, active should be a modifier for the dot class. dot itself shouldn't be a block but rather an element for the slides